src/main/java/frc/robot/subsystems/led/Led2023.java

```diff
@@ -418,275 +476,370 @@ public void setBottom(COLORS_467 color) {
-      if (purpleTimer.hasElapsed(SHOOTING_TIMER_SPEED * i)) {
-        double timeUntilOff = Math.max(0, (SHOOTING_TIMER_SPEED * (i + 2)) - purpleTimer.get());
-        double brightness = (255 * timeUntilOff);
+    public void setColorMovingDown(Color fgColor, Color bgColor) {
+      if (purpleTimer.hasElapsed(
+          SHOOTING_TIMER_SPEED * (RobotConstants.get().led2023LedCount() + 2))) {
+        purpleTimer.reset();
+      }
 
-        if (brightness == 0) {
-          ledStrip.setLED(i, bgColor);
-          ledStrip.update();
+      for (int i = 0; i < RobotConstants.get().led2023LedCount(); i++) {
+        if (purpleTimer.hasElapsed(SHOOTING_TIMER_SPEED * i)) {
+          double timeUntilOff = Math.max(0, (SHOOTING_TIMER_SPEED * (i + 2)) - purpleTimer.get());
+          double brightness = (255 * timeUntilOff);
+
+          if (brightness == 0) {
+            ledStrip.setLED(i, bgColor);
+            ledStrip.update();
+          } else {
+            ledStrip.setRGB(
+                i,
+                (int) (fgColor.red * brightness),
+                (int) (fgColor.green * brightness),
+                (int) (fgColor.blue * brightness));
+            ledStrip.update();
+          }
         } else {
-          ledStrip.setRGB(
-              i,
-              (int) (fgColor.red * brightness),
-              (int) (fgColor.green * brightness),
-              (int) (fgColor.blue * brightness));
+          ledStrip.setLED(i, bgColor);
           ledStrip.update();
         }
-      } else {
-        ledStrip.setLED(i, bgColor);
-        ledStrip.update();
       }
     }
-  }
 
-  public void setColorMovingUp(Color fgColor, Color bgColor) {
-    if (purpleTimer.hasElapsed(
-        SHOOTING_TIMER_SPEED * (RobotConstants.get().led2023LedCount() + 2))) {
-      purpleTimer.reset();
-    }
+    public void setColorMovingUp(Color fgColor, Color bgColor) {
+      if (purpleTimer.hasElapsed(
+          SHOOTING_TIMER_SPEED * (RobotConstants.get().led2023LedCount() + 2))) {
+        purpleTimer.reset();
+      }
 
-    for (int i = 0; i < RobotConstants.get().led2023LedCount(); i++) {
-      int j = RobotConstants.get().led2023LedCount() - i - 1;
-      if (purpleTimer.hasElapsed(SHOOTING_TIMER_SPEED * i)) {
-        double timeUntilOff = Math.max(0, (SHOOTING_TIMER_SPEED * (i + 2)) - purpleTimer.get());
-        double brightness = (255 * timeUntilOff);
+      for (int i = 0; i < RobotConstants.get().led2023LedCount(); i++) {
+        int j = RobotConstants.get().led2023LedCount() - i - 1;
+        if (purpleTimer.hasElapsed(SHOOTING_TIMER_SPEED * i)) {
+          double timeUntilOff = Math.max(0, (SHOOTING_TIMER_SPEED * (i + 2)) - purpleTimer.get());
+          double brightness = (255 * timeUntilOff);
 
-        if (brightness == 0) {
-          ledStrip.setLED(j, bgColor);
-          ledStrip.update();
+          if (brightness == 0) {
+            ledStrip.setLED(j, bgColor);
+            ledStrip.update();
```

Comment:
Remove all calls to ledStrip.update() - you already call it every 20ms with sendData().